app/helpers/automated_tests_helper.rb

```diff
@@ -119,114 +119,52 @@ def create_ant_test_files(assignment)
   # - Process new and updated test files (additional validation to be done at the model level)
   def process_test_form(assignment, params)
 
-    # Hash for storing new and updated test files
     updated_script_files = {}
     updated_support_files = {}
 
-    # Array for checking duplicate file names
-    file_name_array = []
+    testscripts = params[:test_scripts_attributes] || []
+    testsupporters = params[:test_support_files_attributes] || []
 
-    # add existing scripts names
-    params.each do |key, value|
-      unless key[/test_script_\d+/].nil?
-        file_name_array << value
-      end
-    end
+    # Create/Update test scripts
+    testscripts.each do |file_num, file|
+      updated_script_files[file_num] = {}
 
-    # Retrieve all test scripts
-    testscripts = params[:test_scripts_attributes]
-
-    # First check for duplicate script names in test scripts
-    unless testscripts.nil?
-      testscripts.values.each do |tfile|
-        if tfile['script_name'].respond_to?(:original_filename)
-          fname = tfile['script_name'].original_filename
-          # If this is a duplicate script name, raise error and return
-          if !file_name_array.include?(fname)
-            file_name_array << fname
-          else
-            raise I18n.t('automated_tests.duplicate_filename') + fname
-          end
-        end
-      end
-    end
+      # Empty file submission, skip
+      next if testscripts[file_num][:script_name].nil?
```

Comment:
I can't remember if I pointed this out, but on second thought, I'm not sure this check is valid. The form won't have a `:script_name` field any more, since we renamed it.